### DISCUSSION TITLE
Tumblr upgraded to React 17

### DISCUSSION THREAD

**Discussion by blowery** [AprilSylph/XKit-Rewritten#410] [Tech Office]
Hey folks,

FYI, Tumblr just upgraded to React 17, which removes the `__reactInternalInstance` property on DOM elements in favor of `__reactFiber`. It may alter some other React internals that XKit-Rewritten relies on as well, but I have not yet jumped all the way down the rabbit hole.

**Comment by AprilSylph** [comment]
Yep, v0.14.2 tackles the change to `__reactFiber`. Thanks for the message.

**Comment by AprilSylph** [comment]
@blowery I don't think this is directly related, but since you're here... I'm now seeing nonspecific errors when attempting to edit legacy media posts. If I `GET` the NPF representation of a legacy post, edit it, then attempt to `PUT` the edited NPF back, it fails with HTTP 400. Before, there was no issue with editing posts like this. Is this an intended change?

**Reply by blowery** [reply]
I agree, that doesn't sound related to the React upgrade, but I'll ask around.

**Reply by AprilSylph** [reply]
Thank you! Sorry to have bothered you here about it. If it's still occurring tomorrow, I'll send in a proper ticket to support.

**Reply by sndsgd** [reply]
@AprilSylph can you drop the post id in this thread please?

**Reply by AprilSylph** [reply]
@sndsgd Here's the one I first reproduced the error with: https://april-codes.tumblr.com/private/674355161484984320/tumblr_r62htajeUG1t5zyeu

If you need a public post, it's also happening on https://april-codes.tumblr.com/post/674369028333142016

**Reply by sndsgd** [reply]
@AprilSylph FWICT we currently disallow editing legacy posts using NPF if the existing post type is not `answer` or `text`. There are some circumstances that will prevent us from allowing npf edits to legacy `text` posts as well, namely client support and the existence of indented content. The two posts you mentioned above are both `photo` posts, so I can see where/why we're failing the request.

It looks like some of this logic was last tweaked ~5 months ago. Can you please confirm whether you were editing legacy media posts before? And if so, any estimate of the date you were able to successfully edit legacy media posts may be helpful in allow me to sift through changes.

Thanks for reporting and the info :pray:

**Reply by AprilSylph** [reply]
I'm actually _currently_ able to edit these legacy photo posts using NPF:
- https://april-codes.tumblr.com/post/663872293908414464/qatestr-blogless-advertiser-tsp-static
- https://april-codes.tumblr.com/post/659822149266292736
- https://april-codes.tumblr.com/post/642572324409851904/for-qa-beacon-3

What about NPF reblogs of legacy posts? Is it possible something has changed there?

According to the person who reported #411, this only started happening today.

**Reply by sndsgd** [reply]
Looks like the last three posts are actually reblogs that were created using NPF:

| id | is reblog | version | type |
| --- | --- | --- | --- |
| 674369028333142016 | no | legacy | photo |
| 674355161484984320 | yes | legacy | photo |
| 663872293908414464 | yes | npf | photo |
| 659822149266292736 | yes | npf | photoset |
| 642572324409851904 | yes | npf | photoset |

**Reply by AprilSylph** [reply]
I see. That would make sense - I reblogged `674355161484984320` with legacy endpoints to test a different React 17 fix. Perhaps this isn't new, but just hasn't been found until today because XKit Rewritten only creates NPF reblogs. I would be amazed if that were truly the case, but the evidence points towards it being so.
